src/Resolvers/DomainTenantResolver.php

```diff
@@ -28,18 +28,30 @@ class DomainTenantResolver extends Contracts\CachedTenantResolver
 
     public function resolveWithoutCache(...$args): Tenant
     {
-        /** @var Domain $domain */
-        $domain = config('tenancy.domain_model')::where('domain', $args[0])->first();
+        $domain = $args[0];
 
-        if ($domain) {
-            static::$currentDomain = $domain;
+        /** @var Tenant|null $tenant */
+        $tenant = config('tenancy.tenant_model')::query()
+            ->whereHas('domains', function ($query) use ($domain) {
+                $query->select(['tenant_id', 'domain'])->where('domain', $domain);
```

Comment:
To be fair I'd remove the `select()` because any performance gains will be negligible, and it makes the code harder to read. It seems like a microoptimization that will ultimately just lead to issues down the line. For example, someone might use a different column than `tenant_id` — for example, `team_id`. The point of `whereHas()` is that it abstracts these things, and we can safely make as few assumptions as needed, namely that there's a `domain` column, and that there's a defined relationship.

Reply:
Okay, will remove that.